.github/workflows/ci-tests.yml

```diff
@@ -94,6 +94,11 @@ jobs:
       - name: "tests (py${{ matrix.python-version }})"
         env:
           PY_VER: ${{ matrix.python-version }}
-          COVERAGE: ${{ matrix.coverage }}
         run: |
           nox --session tests -- --verbose
+
+      - name: Upload coverage report
+        uses: codecov/codecov-action@v4
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+        if: ${{ matrix.coverage }}
```

Comment:
@stephenworsley For readability, perhaps move the `if:` so that it's after the "uses:" on line 102 ?